_test/test_utilities_herbert/test_sawtooth_iarray.m

```diff
@@ -0,0 +1,110 @@
+classdef test_sawtooth_iarray < TestCase
+    % Test behaviour of Herbert utility function sawtooth_iarray.
+    
+    methods
+        function test_sawtooth_iarray_empty_n (~)
+            % Empty input array, empty n ==> zero(0,1) output
+            n = [];
+            vout = sawtooth_iarray (n);
+            assertEqual(vout, zeros(0,1));
+        end
+        
+        function test_sawtooth_iarray_zero_n (~)
+            % Output should be zero(0,1)
+            n = 0;
+            vout = sawtooth_iarray (n);
+            assertEqual(vout, zeros(0,1));
+        end
+        
+        function test_sawtooth_iarray_scalar_n (~)
+            % Output should be a column independent of shape of input
+            n = 3;
+            vout = sawtooth_iarray (n);
+            vout_ref = [1;2;3];
+            assertEqual(vout, vout_ref);
+        end
+        
+        function test_sawtooth_iarray_row_n (~)
+            % Output should be a column independent of shape of input
+            n = [2,3,1];
+            vout = sawtooth_iarray (n);
+            vout_ref = [1;2;1;2;3;1];
+            assertEqual(vout, vout_ref);
+        end
+        
+        function test_sawtooth_iarray_col_n (~)
+            % Output should be a column independent of shape of input
+            n = [2;3;1];
+            vout = sawtooth_iarray (n);
+            vout_ref = [1;2;1;2;3;1];
+            assertEqual(vout, vout_ref);
+        end
+        
+        function test_sawtooth_iarray_arr_v_arr_n (~)
+            % Output should be a column independent of shape of input
+            n = [2,3,1;...
+                4,5,6];
```

Comment:
allowing this gives rise to an ambiguity in the order of accessing n, this examples could be 2,4,3,5,1,6

Reply:
It is assumed that the order of the elements is always the standard Matlab (and Fortran) order i.e. in this case 2,4,3,5,1,6. This is standard Matlab behaviour.